fem/bilininteg.hpp

```diff
@@ -2732,6 +2805,23 @@ class DGTraceIntegrator : public BilinearFormIntegrator
    void SetupPA(const FiniteElementSpace &fes, FaceType type);
 };
 
+/** Integrator that represents the face terms used for the non-conservative
+    DG discretization of the convection equation:
+    (alpha - 1*sgn(alpha)) < rho_u (u.n) [v],{w} > - < rho_u (u.n) {v},[w] >
+       + beta < rho_u |u.n| [v],[w] >.
+
+    This integrator can be used with alpha=1.0, beta=0.5, together with
+    ConvectionIntegrator to implement an upwind DG discretization in
+    non-conservative form, see ex9 and ex9p. */
+class NonconservativeDGTraceIntegrator : public SumIntegrator
+{
+public:
+   NonconservativeDGTraceIntegrator(VectorCoefficient &u, double a, double b);
```

Comment:
Would `a = -1.0`, `b = -0.5` make sense?

Reply:
`a` should really match `alpha` from the `ConvectionIntegrator`, and the sign of `b` should match the sign of `a`, so setting good default values is a bit tricky.

Reply:
Yes, that's why default values make sense for `NonconservativeDGTraceIntegrator` since it is expected to be used with `NonconservativeDGConvectionIntegrator` as I proposed above.